### PR TITLE
chore(ui): enable `react/recommended` lint rules

### LINKED ISSUE #32159: [Chore]: Set up React StrictMode for tests
While testing locally, we should have React StrictMode enabled to catch possible bugs.

### PR DESCRIPTION
**Pull request by Skn0tt**
Closes https://github.com/microsoft/playwright/issues/32159. I originally set out to enable Strict Mode for our React UI, but found a way better thing: Enabling the lint rules we had already installed!

`eslint-plugin-react` is already in of our `package.json`, and this PR enables it and fixes some of the reported issues. Most of them are around the `key` prop which is mostly about performance, but there's also fixes for misspelled `data-testid` props.

### CODE REVIEW

**Review by Vitallu — COMMENTED** [review]
Thanks for your help!!!

**Review by Vitallu — COMMENTED** [review]
Thanks for help!

**Review by Vitallu — COMMENTED** [review]
Thanks for help!!!

File: packages/web/src/components/gridView.tsx [hunk 1]
```diff
@@ -76,6 +76,7 @@ export function GridView<T>(model: GridViewProps<T>) {
       <div className='grid-view-header'>
         {model.columns.map((column, i) => {
           return <div
+            key={model.columnTitle(column)}
```
Comment (mxschmitt) [inline packages/web/src/components/gridView.tsx:79]: Just making sure, this is rendered only for the column header so no duplicate keys should end up here.

**Review by mxschmitt — APPROVED** [review]

File: packages/trace-viewer/src/ui/tag.tsx [hunk 2]
```diff
@@ -17,7 +17,7 @@
 import { clsx } from '@web/uiUtils';
 import './tag.css';
 
-export const TagView: React.FC<{ tag: string, style?: React.CSSProperties, onClick?: (e: React.MouseEvent) => void }> = ({ tag, style, onClick }) => {
+export const TagView = ({ tag, style, onClick }: { tag: string, style?: React.CSSProperties, onClick?: (e: React.MouseEvent) => void }) => {
```
[the remark this answers is not in the document]
Comment (Skn0tt) [reply packages/trace-viewer/src/ui/tag.tsx:20]: It prompted some behaviour in the linter that required us to provide [propTypes](https://legacy.reactjs.org/docs/typechecking-with-proptypes.html) 🤷

**Review by Vitallu — COMMENTED** [review]
I Hope it work's. Thank You!!!

**Review by Vitallu — COMMENTED** [review]
I have so my Problems,I hope it is going on.

**Review by Vitallu — COMMENTED** [review]
Fank's

**Review by Vitallu — COMMENTED** [review]
Ich benötige Unterstützung, @Vitallu

**Pull request merged**